numpy/core/tests/test_multiarray.py

```diff
@@ -1201,6 +1201,74 @@ def test_count_nonzero_unaligned(self):
             assert_equal(np.count_nonzero(a), builtins.sum(a.tolist()))
 
 
+class TestZeroSizeFlexible(object):
+    @staticmethod
```

Comment:
Is there an advantage to making this a staticmethod?

Reply:
To document that it doesn't use `self`?

Reply:
Well, it does that, don't know if it is worth it as the normal impulse would be to use it only if needed. Then again, my Python experience started when Python 21 was the newest and greatest, so programming conventions may have changed. In fact, it is the latter possibility that interests me. Do you know if it is the case that this sort of programming style is being taught or encouraged?

Reply:
`TestCase` objects (or the inferred use of them by pytest) are pretty weird anyway, because an new one is instantiated for each method call by the test runner. I can change this if you feel strongly.

Reply:
I don't feel strongly about it. On occasion I find your coding style a bit unusual, but not wrong. Just trying to figure out if it is a personal style, or the future of python coding ;)

Reply:
Could easily be personal style. I guess it probably comes from my java background, but even there it might just have been personal style.